GliaWidgetsTests/Sources/GliaTests.swift

```diff
@@ -132,6 +132,78 @@ class GliaTests: XCTestCase {
         XCTAssertTrue(try XCTUnwrap(receivedSceneProvider as? MockedSceneProvider) === expectedSceneProvider)
     }
 
+    func testStartEngagementThrowsErrorWhenEngagementAlreadyExists() throws {
+        var isErrorThrown = false
+        let gliaEnv = Glia.Environment.failing
+        let sdk = Glia(environment: gliaEnv)
+        sdk.rootCoordinator = .mock(engagementKind: .chat, screenShareHandler: .mock)
+
+        let configuration = Configuration(
+            authorizationMethod: .siteApiKey(
+                id: "siteAp1Key",
+                secret: "s3cr3t"
+            ),
+            environment: .beta,
+            site: "site"
+        )
+        try sdk.configure(with: configuration, queueId: "queueID")
+
+        do {
+            try sdk.startEngagement(engagementKind: .chat)
+        } catch GliaError.engagementExists {
+            isErrorThrown = true
+        } catch {
+            XCTFail()
+        }
+
+        XCTAssertTrue(isErrorThrown)
```

Comment:
N.B. You can also use `XCTAssertThrowsError`.

Reply:
Fixed 69dceb129ca16184df4581dec6e62fd7900c1810. Now it looks significantly better. Thank you!